pkg/operator/apis/monitoring/v1/pod_config.go

```diff
@@ -382,6 +386,76 @@ func relabelingsForMetadata(keys map[string]struct{}) (res []*relabel.Config) {
 			TargetLabel:  "node",
 		})
 	}
+	if _, ok := keys["top_level_controller_name"]; ok {
```

Comment:
I would say this deserves a simple unit test that puts what we expect to see in those `__meta` labels and do the resulting relabelling. We can then track regressions or weird __meta label states over time. WDYT? (: